### PR TITLE
add knownIdentities cache

### PR DESCRIPTION
**Pull request by shamb0t**
Adds cache for known identities

### CODE REVIEW

File: src/identities.js [hunk 1]
```diff
@@ -64,15 +70,47 @@ class Identities {
...
+
+    const knownID = this._knownIdentities.get(identity.signatures.id)
+    if (knownID) {
+      return identity.id === knownID.id &&
+             identity.publicKey === knownID.publicKey &&
+             identity.signatures.id === knownID.signatures.id &&
+             identity.signatures.publicKey === knownID.signatures.publicKey
+    }
+
+    const verifyIdSig = await Keystore.verify(
```
Comment (haadcode) [inline src/identities.js]: Which `Keystore` instance does this use? Shouldn't this be `this._keystore.verify()`?

File: src/identities.js [hunk 2]
```diff
@@ -64,15 +70,47 @@ class Identities {
...
+
+    return verified
   }
 
   static async verifyIdentity (identity) {
+    if (!Identity.isIdentity(identity)) {
+      return false
+    }
+
+    const verifyIdSig = await Keystore.verify(
```
Comment (haadcode) [inline src/identities.js]: Which `Keystore` instance (or data directory) does this use?

Comment (shamb0t) [reply src/identities.js]: @haadcode it uses the Keystore's static `verify` method, it doesn't use a directory  https://github.com/orbitdb/orbit-db-keystore/blob/409d6d46f1260c2a00ae2920a068460a2298481c/src/keystore.js#L182

Comment (haadcode) [reply src/identities.js]: Thanks @shamb0t, makes sense 👍

Comment (shamb0t) [reply src/identities.js]: @haadcode hmm after running tests with orbit-db its true that if a custom keystore is used then the custom keystore should also be used for verifying. Does it make sense that the `static verifyIdentity` would only verify identities which used `orbit-db-keystore`? (ie `this._keystore.verify` in the non-static method and `Keystore.verify` in the static method)

Comment (haadcode) [reply src/identities.js]: You're right, perhaps we don't need the static method. I'd keep it in though in this PR and remove it later on. For now, we should be good, right?

**Comment by haadcode** [comment]
LGTM @shamb0t 👍

**Pull request merged**